airflow/providers/amazon/aws/log/s3_task_handler.py

```diff
@@ -178,7 +180,13 @@ def s3_read(self, remote_log_location: str, return_error: bool = False) -> str:
         :return: the log found at the remote_log_location
         """
         try:
-            return self.hook.read_key(remote_log_location)
+            range: str = None
+            if page_number is not None:
+                page_size = 1024 * 100  # TODO: Create config for page_size
```

Comment:
I wonder if we should just put this in the API interface. So instead of a page number (of fixed size pages), the API user would do `offset=0&limit=500` and `offset=500&limit=500` and so on. (I took the argument names from SQL but other suggestions are very much welcomed.)

Reply:
Does it even need to be API paremeters, or could we do what S3 does, and do this as an HTTP Range request?

Reply:
> Does it even need to be API paremeters, or could we do what S3 does, and do this as an HTTP Range request?

This is currently the plan

Reply:
I think what Ash means is we take the Range header in the API, and forward it to the log server.